controller/instance_manager_controller.go

```diff
@@ -612,6 +617,12 @@ func (imc *InstanceManagerController) handlePod(im *longhorn.InstanceManager) er
 		return err
 	}
 
+	if isPodDeletedOrNotRunning {
+		if err := imc.cleanupInstanceManagerOrphanInstances(im); err != nil {
```

Comment:
The function `cleanupInstanceManagerOrphanInstances` actually deletes Orphan resources rather than instances. Suggest renaming to `deleteOrphans(im)`.

```suggestion
		if err := imc.deleteOrphans(im); err != nil {
```